plugins/locale.js

```diff
@@ -1,25 +1,62 @@
-import { join } from "node:path";
-import { existsSync, readFileSync } from "node:fs";
+import { existsSync } from "node:fs";
 import fp from "fastify-plugin";
 import chalk from "chalk";
 import emojiFlags from "emoji-flags";
+import { join } from "node:path";
+import PathResolver from "../lib/path.js";
+
+/**
+ * Find emoji flag entry based on country name rather than code
+ * @param {string} countryName
+ * @returns
+ */
+function emojiDataLookup(countryName) {
+  return emojiFlags.data.find((entry) => entry.name === countryName);
+}
 
-// function that takes a locale string and returns the emoji flag for that locale
-// handles formats like en-US, en, en-US-POSIX
-// map countries to languages
+/**
+ * This function does a best effort lookup of a locales ISO 3166-1 alpha-2 country code
+ * and returns the corresponding emoji flag.
+ * @param {string} locale the currently assigned locale
+ * @returns {string} the emoji flag
+ */
 function getFlag(locale) {
   if (!locale) return "";
-  const countryCode = locale.split("-")[1];
-  if (!countryCode) return "";
-  return `${emojiFlags.countryCode(countryCode).emoji}  `;
+
+  const countryNames = new Intl.DisplayNames("en", { type: "region" });
+  const [first, second] = locale.split("-").map((part) => part?.toLocaleUpperCase());
+
+  const [firstCountryName, secondCountryName] = [countryNames.of(first ?? "EN"), countryNames.of(second ?? "EN")];
+
+  if (first && firstCountryName !== first) {
+    return `${emojiDataLookup(firstCountryName)?.emoji}  `;
+  } else if (second && secondCountryName !== second) {
+    return `${emojiDataLookup(secondCountryName)?.emoji}  `;
+  } else {
+    switch (locale) {
+      case "en":
+        return `${emojiFlags.countryCode("US")?.emoji}  `;
+      case "nb":
+      case "nn":
```

Comment:
hmm, trouble with all cases is, where do you stop? 🤔

Reply:
> 

@digitalsadhu nobody knows 🤷 Just noticed that what we recommend/mention in the RFC wans't picked up properly here.